src/Ast.ml

```diff
@@ -50,6 +50,14 @@ let is_kwdopchar = function
       true
   | _ -> false
 
+let is_monadic_last_char = function '=' | '|' | '>' -> true | _ -> false
+
+let is_monadic_id s =
```

Comment:
I don't think we should use the operators characters to guess that. This is accepting operators like `||` and `:=` and it's rejecting a lot of potential monadic operators.